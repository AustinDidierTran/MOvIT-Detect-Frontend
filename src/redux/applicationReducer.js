// ------------------------------------
// Constants
// ------------------------------------

let url;

switch (process.env.NODE_ENV) {
  case 'production':
  case 'pi':
    url = 'http://192.168.4.1:1880/';
    break;
  case 'local':
    url = 'http://localhost:1880/';
    break;
  default:
    url = 'https://private-f2484-movitplus.apiary-mock.com/';
}

export const URL = url;
export const LANGUAGE = 'LANGUAGE';
export const FR = 'FR';
export const EN = 'EN';
export const PROFILE = 'PROFILE';
export const TOKEN = 'TOKEN';

// ------------------------------------
// Actions
// ------------------------------------
function changeLanguage() {
  return {
    type: LANGUAGE,
  };
}

function changeProfile(profileName) {
  return {
    type: PROFILE,
    profile: profileName,
  };
}

function changeToken(tokenString) {
  return {
    type: TOKEN,
    token: tokenString,
  };
}

export const ApplicationActions = {
  changeLanguage,
  changeProfile,
  changeToken,
};
// ------------------------------------
// Action Handlers
// ------------------------------------
const ACTION_HANDLERS = {
  [LANGUAGE]: state => (
    { ...state, language: state.language === FR ? EN : FR }
  ),
  [PROFILE]: (state, action) => (
    { ...state, profile: action.profile }
  ),
  [TOKEN]: (state, action) => (
    { ...state, token: action.token, header: { headers: { Authorization: action.token } } }
  ),
};

// ------------------------------------
// Reducer
// ------------------------------------

export const initApplication = {
  language: 'FR',
  profile: '',
  userName: '',
  userID: '',
  token: null,
  header: {},
  maxAngle: null,
  userWeight: null,
};
export default function applicationReducer(state = initApplication, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
