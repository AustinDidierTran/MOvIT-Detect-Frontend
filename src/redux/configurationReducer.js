export const DATA_AGREEMENT = 'DATA_AGREEMENT';
export const LIGHT_AGREEMENT = 'LIGHT_AGREEMENT';
export const NOTIFICATION_AGREEMENT = 'NOTIFICATION_AGREEMENT';
export const DATA_DISAGREE_PERDIOD = 'DATA_DISAGREE_PERDIOD';
export const LIGHT_DISAGREE_PERDIOD = 'LIGHT_DISAGREE_PERDIOD';
export const NOTIFICATION_DISAGREE_PERDIOD = 'NOTIFICATION_DISAGREE_PERDIOD';

// -------------- CHECKBOX --------------
function changeDataAgreement() {
  return {
    type: DATA_AGREEMENT,
  };
}
function changeLightAgreement() {
  return {
    type: LIGHT_AGREEMENT,
  };
}
function changeNotificationAgreement() {
  return {
    type: NOTIFICATION_AGREEMENT,
  };
}

// -------------- TEXT --------------
function changeDataDisagreePeriod(period) {
  return {
    type: DATA_DISAGREE_PERDIOD,
    dataDisagreePeriod: period,
  };
}
function changeLightDisagreePeriod(period) {
  return {
    type: LIGHT_DISAGREE_PERDIOD,
    lightDisagreePeriod: period,
  };
}
function changeNotificationDisagreePeriod(period) {
  return {
    type: NOTIFICATION_DISAGREE_PERDIOD,
    notificationDisagreePeriod: period,
  };
}

export const ConfigurationActions = {
  changeDataAgreement,
  changeLightAgreement,
  changeNotificationAgreement,
  changeDataDisagreePeriod,
  changeLightDisagreePeriod,
  changeNotificationDisagreePeriod,
};

const ACTION_HANDLERS = {
  [DATA_AGREEMENT]: state => (
    { ...state, dataAgreement: !state.dataAgreement }
  ),
  [LIGHT_AGREEMENT]: state => (
    { ...state, lightAgreement: !state.lightAgreement }
  ),
  [NOTIFICATION_AGREEMENT]: state => (
    { ...state, notificationAgreement: !state.notificationAgreement }
  ),
  [DATA_DISAGREE_PERDIOD]: (state, action) => (
    { ...state, dataDisagreePeriod: action.dataDisagreePeriod }
  ),
  [LIGHT_DISAGREE_PERDIOD]: (state, action) => (
    { ...state, lightDisagreePeriod: action.lightDisagreePeriod }
  ),
  [NOTIFICATION_DISAGREE_PERDIOD]: (state, action) => (
    { ...state, notificationDisagreePeriod: action.notificationDisagreePeriod }
  ),
};

export const initConfiguration = {
  dataAgreement: true,
  lightAgreement: true,
  notificationAgreement: true,
  dataDisagreePeriod: null,
  lightDisagreePeriod: null,
  notificationDisagreePeriod: null,
};

export default function applicationReducer(state = initConfiguration, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
