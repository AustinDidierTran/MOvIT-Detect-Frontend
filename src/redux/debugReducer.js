/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

export const IS_LED_BLINKING_ENABLED = 'IS_LED_BLINKING_ENABLED';
export const IS_VIBRATION_ENABLED = 'IS_VIBRATION_ENABLED';

// -------------- CHECKBOX --------------
function changeIsLedBlinkingEnabled(isEnabled) {
  return {
    type: IS_LED_BLINKING_ENABLED,
    isLedBlinkingEnabled: isEnabled,
  };
}
function changeIsVibrationEnabled(isEnabled) {
  return {
    type: IS_VIBRATION_ENABLED,
    isVibrationEnabled: isEnabled,
  };
}

export const DebugActions = {
  changeIsLedBlinkingEnabled,
  changeIsVibrationEnabled,
};

const ACTION_HANDLERS = {
  [IS_LED_BLINKING_ENABLED]: (state, action) => (
    { ...state, isLedBlinkingEnabled: action.isLedBlinkingEnabled }
  ),
  [IS_VIBRATION_ENABLED]: (state, action) => (
    { ...state, isVibrationEnabled: action.isVibrationEnabled }
  ),
};

export const initParameter = {
  isLedBlinkingEnabled: true,
  isVibrationEnabled: true,
};

export default function applicationReducer(state = initParameter, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
