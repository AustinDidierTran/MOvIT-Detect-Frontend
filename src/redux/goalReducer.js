/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin Didier Tran
 */

export const TILT_FREQUENCY_GOAL = 'TILT_FREQUENCY_GOAL';
export const TILT_LENGTH_GOAL = 'TILT_LENGTH_GOAL';
export const TILT_ANGLE_GOAL = 'TILT_ANGLE_GOAL';


// -------------- SLIDERS --------------

function changeTiltFrequencyGoal(tiltFreq) {
  return {
    type: TILT_FREQUENCY_GOAL,
    tiltFrequencyGoal: tiltFreq,
  };
}
function changeTiltLengthGoal(tiltLen) {
  return {
    type: TILT_LENGTH_GOAL,
    tiltLengthGoal: tiltLen,
  };
}
function changeTiltAngleGoal(tiltAng) {
  return {
    type: TILT_ANGLE_GOAL,
    tiltAngleGoal: tiltAng,
  };
}

export const GoalActions = {
  changeTiltFrequencyGoal,
  changeTiltLengthGoal,
  changeTiltAngleGoal,
};

const ACTION_HANDLERS = {
  [TILT_FREQUENCY_GOAL]: (state, action) => (
    { ...state, tiltFrequencyGoal: action.tiltFrequencyGoal }
  ),
  [TILT_LENGTH_GOAL]: (state, action) => (
    { ...state, tiltLengthGoal: action.tiltLengthGoal }
  ),
  [TILT_ANGLE_GOAL]: (state, action) => (
    { ...state, tiltAngleGoal: action.tiltAngleGoal }
  ),
};

// ------------------------------------
// Reducer
// ------------------------------------

export const initGoal = {
  tiltFrequencyGoal: 0,
  tiltLengthGoal: 0,
  tiltAngleGoal: 0,
};

export default function applicationReducer(state = initGoal, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}

