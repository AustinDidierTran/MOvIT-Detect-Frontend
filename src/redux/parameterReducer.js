/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin Didier Tran
 */

export const DATA_AGREEMENT = 'DATA_AGREEMENT';
export const LIGHT_AGREEMENT = 'LIGHT_AGREEMENT';
export const NOTIFICATION_AGREEMENT = 'NOTIFICATION_AGREEMENT';
export const DATA_DISAGREE_PERIOD = 'DATA_DISAGREE_PERDIOD';
export const LIGHT_DISAGREE_PERIOD = 'LIGHT_DISAGREE_PERDIOD';
export const NOTIFICATION_DISAGREE_PERDIOD = 'NOTIFICATION_DISAGREE_PERDIOD';

// -------------- CHECKBOX --------------
function changeDataAgreement(agree) {
  return {
    type: DATA_AGREEMENT,
    dataAgreement: agree,
  };
}
function changeLightAgreement(agree) {
  return {
    type: LIGHT_AGREEMENT,
    lightAgreement: agree,
  };
}
function changeNotificationAgreement(agree) {
  return {
    type: NOTIFICATION_AGREEMENT,
    notificationAgreement: agree,
  };
}

// -------------- TEXT --------------
function changeDataDisagreePeriod(period) {
  return {
    type: DATA_DISAGREE_PERIOD,
    dataDisagreePeriod: period,
  };
}
function changeLightDisagreePeriod(period) {
  return {
    type: LIGHT_DISAGREE_PERIOD,
    lightDisagreePeriod: period,
  };
}
function changeNotificationDisagreePeriod(period) {
  return {
    type: NOTIFICATION_DISAGREE_PERDIOD,
    notificationDisagreePeriod: period,
  };
}

export const ParameterActions = {
  changeDataAgreement,
  changeLightAgreement,
  changeNotificationAgreement,
  changeDataDisagreePeriod,
  changeLightDisagreePeriod,
  changeNotificationDisagreePeriod,
};

const ACTION_HANDLERS = {
  [DATA_AGREEMENT]: (state, action) => (
    { ...state, dataAgreement: action.dataAgreement }
  ),
  [LIGHT_AGREEMENT]: (state, action) => (
    { ...state, lightAgreement: action.lightAgreement }
  ),
  [NOTIFICATION_AGREEMENT]: (state, action) => (
    { ...state, notificationAgreement: action.notificationAgreement }
  ),
  [DATA_DISAGREE_PERIOD]: (state, action) => (
    { ...state, dataDisagreePeriod: action.dataDisagreePeriod }
  ),
  [LIGHT_DISAGREE_PERIOD]: (state, action) => (
    { ...state, lightDisagreePeriod: action.lightDisagreePeriod }
  ),
  [NOTIFICATION_DISAGREE_PERDIOD]: (state, action) => (
    { ...state, notificationDisagreePeriod: action.notificationDisagreePeriod }
  ),
};

export const initParameter = {
  dataAgreement: true,
  lightAgreement: true,
  notificationAgreement: true,
  dataDisagreePeriod: null,
  lightDisagreePeriod: null,
  notificationDisagreePeriod: null,
};

export default function applicationReducer(state = initParameter, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
