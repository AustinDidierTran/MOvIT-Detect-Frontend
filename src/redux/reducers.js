/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin Didier Tran
 */

import { combineReducers } from 'redux';
import applicationReducer from './applicationReducer';
import recommendationReducer from './recommendationReducer';
import goalReducer from './goalReducer';
import parameterReducer from './parameterReducer';
import configurationReducer from './configurationReducer';

export default combineReducers({
  applicationReducer,
  recommendationReducer,
  goalReducer,
  parameterReducer,
  configurationReducer,
});
