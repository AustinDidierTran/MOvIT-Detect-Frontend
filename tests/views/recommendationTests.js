/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import MockAdapter from 'axios-mock-adapter';
import PropTypes from 'prop-types';
import React from 'react';
import axios from 'axios';
import configureMockStore from 'redux-mock-store';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import { URL } from '../../src/redux/applicationReducer';
import Recommendation from '../../src/views/recommendation';

Enzyme.configure({ adapter: new Adapter() });

const TIMEOUT = 1000;

function initializeMockAdapter() {
  const mock = new MockAdapter(axios);
  const response = true;

  mock.onGet(`${URL}recommandation`).reply(200, response);
}

describe('Recommendation Tests', () => {
  let wrapper;
  const initialState = {
    applicationReducer: {
      language: 'FR',
    },
    recommendationReducer: {
      tiltFrequencyWeight: 10,
      tiltLengthWeight: 12,
      tiltAngleWeight: 20,
      tiltAngleMoving: 0,
      tiltAngleRest: 0,
      painRecommentation: '',
      swellingRecommendation: '',
      restRecommendation: '',
      transferRecommendation: '',
      comfortRecommendation: '',
      otherRecommendations: '',
      otherRecommendationsTitle: '',
      reduceWeight: true,
      reduceSwelling: true,
      reduceSlidingMoving: true,
      reduceSlidingRest: true,
      reducePain: true,
      allowRest: true,
      easeTransfer: true,
      improveComfort: true,
      other: true,
    },
    configurationReducer: {
      maxAngle: 60,
    },
  };
  const secToMin = 60;
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    history: [],
  };

  initializeMockAdapter();

  beforeEach(() => {
    wrapper = shallow(<Recommendation store={store} {...props} />).dive();
    wrapper.setState({ isLoaded: true, hasErrors: false });

    const spy = sinon.spy(wrapper.instance(), 'load');

    setTimeout(() => {
      expect(spy.calledOnce).toEqual(true);
      expect(wrapper.state('maxSliderAngle')).toEqual(90);
      expect(wrapper.state('isLoaded')).toEqual(true);
      expect(wrapper.state('hasErrors')).toEqual(false);
    }, TIMEOUT);
  });

  it('should have proptypes', () => {
    const actualValue = Recommendation.WrappedComponent.propTypes;

    const expectedValue = {
      history: PropTypes.object.isRequired,
      language: PropTypes.string.isRequired,
      swellingRecommendation: PropTypes.string,
      painRecommendation: PropTypes.string,
      restRecommendation: PropTypes.string,
      transferRecommendation: PropTypes.string,
      comfortRecommendation: PropTypes.string,
      otherRecommendations: PropTypes.string,
      maxAngle: PropTypes.number,
      reduceWeight: PropTypes.bool,
      tiltFrequencyWeight: PropTypes.number.isRequired,
      tiltLengthWeight: PropTypes.number.isRequired,
      tiltAngleWeight: PropTypes.number.isRequired,
      changeTiltFrequencyGoal: PropTypes.func.isRequired,
      changeTiltLengthGoal: PropTypes.func.isRequired,
      reduceSlidingMoving: PropTypes.bool.isRequired,
      tiltAngleMoving: PropTypes.number.isRequired,
      tiltAngleRest: PropTypes.number.isRequired,
      allowRest: PropTypes.bool.isRequired,
      easeTransfers: PropTypes.bool.isRequired,
      improveComfort: PropTypes.bool.isRequired,
      other: PropTypes.bool.isRequired,
      otherRecommendationsTitle: PropTypes.string,
      reduceSlidingRest: PropTypes.bool.isRequired,
      reduceSwelling: PropTypes.bool.isRequired,
      reducePain: PropTypes.bool.isRequired,
      changeTiltAngleGoal: PropTypes.func.isRequired,
      changeTiltFrequencyWeight: PropTypes.func.isRequired,
      changeTiltAngleWeight: PropTypes.func.isRequired,
      changeTiltLengthWeight: PropTypes.func.isRequired,
      changeReduceWeight: PropTypes.func.isRequired,
      changeReduceSlidingMoving: PropTypes.func.isRequired,
      changeTiltAngleMoving: PropTypes.func.isRequired,
      changeReduceSlidingRest: PropTypes.func.isRequired,
      changeTiltAngleRest: PropTypes.func.isRequired,
      changeReduceSwelling: PropTypes.func.isRequired,
      otherRecommendationTitle: PropTypes.func,
      reduceSwellingRecommendation: PropTypes.func,
      changeImproveComfort: PropTypes.func,
      improveComfortRecommendation: PropTypes.func,
      changeReducePain: PropTypes.func,
      otherRecommendation: PropTypes.func,
      reducePainRecommendation: PropTypes.func,
      changeOther: PropTypes.func,
      easeTransfersRecommendation: PropTypes.func,
      changeEaseTransfers: PropTypes.func,
      changeAllowRest: PropTypes.func,
      allowRestRecommendation: PropTypes.func,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should map the data with the response', async () => {
    const spy = sinon.spy(wrapper.instance(), 'mapData');

    await wrapper.instance().load();

    expect(spy.getCalls()[0].args[0]).toEqual(true);
    expect(wrapper.state('isLoaded')).toEqual(true);
    expect(wrapper.state('hasErrors')).toEqual(false);
  });

  it('should map the data', () => {
    const response = {
      reduceWeight: {
        tiltAngle: 40,
        tiltFrequency: 100,
        tiltLength: 6,
      },
      reduceSlidingMoving: 30,
      reduceSlidingRest: 46,
      reduceSwelling: 'Do something new',
      reducePain: 'Tilt as needed',
      allowRest: 'yes',
      easeTransfers: 'Tilt as needed',
      improveComfort: 'Tilt as needed',
      other: {
        title: 'New Recommendation Title',
        value: 'New Recommendation Description',
      },
    };

    store.clearActions();

    wrapper.instance().mapData(response);

    const actions = store.getActions();

    expect(actions[0].type).toEqual('REDUCE_WEIGHT');
    expect(actions[1].type).toEqual('TILT_FREQUENCY_WEIGHT');
    expect(actions[2].type).toEqual('TILT_LENGTH_WEIGHT');
    expect(actions[3].type).toEqual('TILT_ANGLE_WEIGHT');
    expect(actions[4].type).toEqual('REDUCE_SLIDING_MOVING');
    expect(actions[5].type).toEqual('TILT_ANGLE_MOVING');
    expect(actions[6].type).toEqual('REDUCE_SLIDING_REST');
    expect(actions[7].type).toEqual('TILT_ANGLE_REST');
    expect(actions[8].type).toEqual('REDUCE_SWELLING');
    expect(actions[9].type).toEqual('REDUCE_SWELLING_RECOMMENDATION');
    expect(actions[10].type).toEqual('REDUCE_PAIN');
    expect(actions[11].type).toEqual('REDUCE_PAIN_RECOMMENDATION');
    expect(actions[12].type).toEqual('ALLOW_REST');
    expect(actions[13].type).toEqual('ALLOW_REST_RECOMMENDATION');
    expect(actions[14].type).toEqual('EASE_TRANSFERS');
    expect(actions[15].type).toEqual('EASE_TRANSFERS_RECOMMENDATION');
    expect(actions[16].type).toEqual('IMPROVE_COMFORT');
    expect(actions[17].type).toEqual('IMPROVE_COMFORT_RECOMMENDATION');
    expect(actions[18].type).toEqual('OTHER');
    expect(actions[19].type).toEqual('OTHER_RECOMMENDATION_TITLE');
    expect(actions[20].type).toEqual('OTHER_RECOMMENDATION');

    expect(actions[0].reduceWeight).toEqual(true);
    expect(actions[1].tiltFrequencyWeight).toEqual(100/secToMin);
    expect(actions[2].tiltLengthWeight).toEqual(6/secToMin);
    expect(actions[3].tiltAngleWeight).toEqual(40);
    expect(actions[4].reduceSlidingMoving).toEqual(true);
    expect(actions[5].tiltAngleMoving).toEqual(30);
    expect(actions[6].reduceSlidingRest).toEqual(true);
    expect(actions[7].tiltAngleRest).toEqual(46);
    expect(actions[8].reduceSwelling).toEqual(true);
    expect(actions[9].swellingRecommendation).toEqual('Do something new');
    expect(actions[10].reducePain).toEqual(true);
    expect(actions[11].painRecommendation).toEqual('Tilt as needed');
    expect(actions[12].allowRest).toEqual(true);
    expect(actions[13].restRecommendation).toEqual('yes');
    expect(actions[14].easeTransfers).toEqual(true);
    expect(actions[15].transferRecommendation).toEqual('Tilt as needed');
    expect(actions[16].improveComfort).toEqual(true);
    expect(actions[17].comfortRecommendation).toEqual('Tilt as needed');
    expect(actions[18].other).toEqual(true);
    expect(actions[19].otherRecommendationsTitle).toEqual('New Recommendation Title');
    expect(actions[20].otherRecommendations).toEqual('New Recommendation Description');
  });

  it('should change the tilt frequency', () => {
    store.clearActions();

    wrapper.instance().changeTiltFrequency(28);

    const actions = store.getActions();

    expect(actions[0].type).toEqual('TILT_FREQUENCY_WEIGHT');
    expect(actions[1].type).toEqual('TILT_FREQUENCY_GOAL');
    expect(actions[0].tiltFrequencyWeight).toEqual(28);
    expect(actions[1].tiltFrequencyGoal).toEqual(28);
  });

  it('should change the tilt length', () => {
    store.clearActions();

    wrapper.instance().changeTiltLength(33);

    const actions = store.getActions();

    expect(actions[0].type).toEqual('TILT_LENGTH_WEIGHT');
    expect(actions[1].type).toEqual('TILT_LENGTH_GOAL');
    expect(actions[0].tiltLengthWeight).toEqual(33);
    expect(actions[1].tiltLengthGoal).toEqual(33);
  });

  it('should change the tilt angle', () => {
    store.clearActions();

    wrapper.instance().changeTiltAngle(60);

    const actions = store.getActions();

    expect(actions[0].type).toEqual('TILT_ANGLE_WEIGHT');
    expect(actions[1].type).toEqual('TILT_ANGLE_GOAL');
    expect(actions[0].tiltAngleWeight).toEqual(60);
    expect(actions[1].tiltAngleGoal).toEqual(60);
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
