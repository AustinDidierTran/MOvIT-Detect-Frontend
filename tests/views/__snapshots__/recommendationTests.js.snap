// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommendation Tests should match the snapshot 1`] = `
<div
  className="mt-5"
>
  <div
    className="container"
  >
    <center>
      <h2>
        recommendations.FR
      </h2>
    </center>
    <legend
      className="text-center header"
    >
      <h4>
        recommendations.recommendationsText.FR
      </h4>
    </legend>
    <div>
      <div
        className="pt-2 pl-3 mt-1"
      >
        <Checkbox
          checked={true}
          className={null}
          disabled={false}
          id={null}
          inputId="reduceWeightCheck"
          label="Reduce weight"
          name={null}
          onChange={[Function]}
          onContextMenu={null}
          onMouseDown={null}
          readOnly={false}
          style={null}
          value={null}
        />
        <label
          htmlFor="reduceWeightCheck"
          id="test"
        >
          recommendations.reduceWeight.FR
        </label>
        <i
          className="fa fa-info-circle pl-2"
          id="reduceWeightInfo"
        />
        <Tooltip
          appendTo="body"
          escape={true}
          for="#reduceWeightInfo"
          hideDelay={null}
          onBeforeShow={null}
          positionStyle={null}
          showDelay={null}
          title="recommendations.reduceWeight.tooltip.FR"
          tooltipDisabled={false}
          tooltipEvent="hover"
          tooltipPosition="right"
          tooltipStyleClass={null}
        />
        <Connect(TiltSliders)
          maxAngle={60}
          onAngleChange={[Function]}
          onFrequencyChange={[Function]}
          onLengthChange={[Function]}
          tiltAngle={20}
          tiltFrequency={10}
          tiltLength={12}
        />
      </div>
      <Connect(AngleRecommendation)
        id="slidingMoving"
        maxAngle={60}
        onChangeActive={[Function]}
        onChangeValue={[Function]}
        recActive={true}
        title="recommendations.slidingMoving.FR"
        tooltip="recommendations.slidingMoving.tooltip.FR"
        value={0}
      />
      <Connect(AngleRecommendation)
        id="slidingRest"
        maxAngle={60}
        onChangeActive={[Function]}
        onChangeValue={[Function]}
        recActive={true}
        title="recommendations.slidingRest.FR"
        tooltip="recommendations.slidingRest.tooltip.FR"
        value={0}
      />
      <Connect(TextRecommendation)
        id="reduceSwelling"
        onChangeActive={[Function]}
        onChangeValue={[Function]}
        recActive={true}
        title="recommendations.reduceSwelling.FR"
        tooltip="recommendations.reduceSwelling.tooltip.FR"
        value=""
      />
      <Connect(TextRecommendation)
        id="reducePain"
        onChangeActive={[Function]}
        onChangeValue={[Function]}
        recActive={true}
        title="recommendations.reducePain.FR"
        tooltip="recommendations.reducePain.tooltip.FR"
        value=""
      />
      <Connect(TextRecommendation)
        id="allowRest"
        onChangeActive={[Function]}
        onChangeValue={[Function]}
        recActive={true}
        title="recommendations.rest.FR"
        tooltip="recommendations.rest.tooltip.FR"
      />
      <Connect(TextRecommendation)
        id="easeTransfer"
        onChangeActive={[Function]}
        onChangeValue={[Function]}
        title="recommendations.transfer.FR"
        tooltip="recommendations.transfer.tooltip.FR"
        value=""
      />
      <Connect(TextRecommendation)
        id="improveComfort"
        onChangeActive={[Function]}
        onChangeValue={[Function]}
        recActive={true}
        title="recommendations.comfort.FR"
        tooltip="recommendations.comfort.tooltip.FR"
        value=""
      />
      <Connect(OtherRecommendation)
        id="other"
        onChangeActive={[Function]}
        onChangeRecTitle={[Function]}
        onChangeValue={[Function]}
        recActive={true}
        recTitle=""
        title="recommendations.other.FR"
        tooltip="recommendations.other.tooltip.FR"
        value=""
      />
      <Connect(SumbitButtons)
        onCancel={[Function]}
        onSave={[Function]}
      />
    </div>
  </div>
</div>
`;
